databricks/koalas/tests/test_groupby.py

```diff
@@ -139,53 +139,84 @@ def test_groupby_simple(self):
 
+        # non-string names
+        pdf = pd.DataFrame(
+            {
+                10: [1, 2, 6, 4, 4, 6, 4, 3, 7],
+                20: [4, 2, 7, 3, 3, 1, 1, 1, 2],
+                30: [4, 2, 7, 3, None, 1, 1, 1, 2],
+                40: list("abcdefght"),
+            },
+            index=[0, 1, 3, 5, 6, 8, 9, 9, 9],
+        )
+        kdf = ks.from_pandas(pdf)
+
+        for as_index in [True, False]:
+            if as_index:
+                sort = lambda df: df.sort_index()
+            else:
+                sort = lambda df: df.sort_values(10).reset_index(drop=True)
+            self.assert_eq(
+                sort(kdf.groupby(10, as_index=as_index).sum()),
+                sort(pdf.groupby(10, as_index=as_index).sum()),
+            )
+            self.assert_eq(
+                sort(kdf.groupby(10, as_index=as_index)[20].sum()),
+                sort(pdf.groupby(10, as_index=as_index)[20].sum()),
+            )
+            self.assert_eq(
+                sort(kdf.groupby(10, as_index=as_index)[[20, 30]].sum()),
+                sort(pdf.groupby(10, as_index=as_index)[[20, 30]].sum()),
+            )
+
     def test_groupby_multiindex_columns(self):
         pdf = pd.DataFrame(
             {
-                ("x", "a"): [1, 2, 6, 4, 4, 6, 4, 3, 7],
-                ("x", "b"): [4, 2, 7, 3, 3, 1, 1, 1, 2],
-                ("y", "c"): [4, 2, 7, 3, None, 1, 1, 1, 2],
-                ("z", "d"): list("abcdefght"),
+                (10, "a"): [1, 2, 6, 4, 4, 6, 4, 3, 7],
+                (10, "b"): [4, 2, 7, 3, 3, 1, 1, 1, 2],
+                (20, "c"): [4, 2, 7, 3, None, 1, 1, 1, 2],
+                (30, "d"): list("abcdefght"),
             },
             index=[0, 1, 3, 5, 6, 8, 9, 9, 9],
         )
         kdf = ks.from_pandas(pdf)
 
         self.assert_eq(
-            kdf.groupby(("x", "a")).sum().sort_index(), pdf.groupby(("x", "a")).sum().sort_index()
+            kdf.groupby((10, "a")).sum().sort_index(), pdf.groupby((10, "a")).sum().sort_index()
         )
         self.assert_eq(
-            kdf.groupby(("x", "a"), as_index=False)
+            kdf.groupby((10, "a"), as_index=False)
             .sum()
-            .sort_values(("x", "a"))
+            .sort_values((10, "a"))
             .reset_index(drop=True),
-            pdf.groupby(("x", "a"), as_index=False)
+            pdf.groupby((10, "a"), as_index=False)
             .sum()
-            .sort_values(("x", "a"))
+            .sort_values((10, "a"))
             .reset_index(drop=True),
         )
         self.assert_eq(
-            kdf.groupby(("x", "a"))[[("y", "c")]].sum().sort_index(),
-            pdf.groupby(("x", "a"))[[("y", "c")]].sum().sort_index(),
+            kdf.groupby((10, "a"))[[(20, "c")]].sum().sort_index(),
+            pdf.groupby((10, "a"))[[(20, "c")]].sum().sort_index(),
+        )
+
+        # TODO: a pandas bug?
+        #  expected = pdf.groupby((10, "a"))[(20, "c")].sum().sort_index()
```

Comment:
nit of nit: one more space here in the comment 😅

Reply:
Yes, and I think It's pandas bug. I'll report to their repository

Reply:
Anyway, `pdf.groupby((10, "a"))[[(20, "c")]].sum()` seems working.

```python
>>> pdf
  10      20 30
   a  b    c  d
0  1  4  4.0  a
1  2  2  2.0  b
3  6  7  7.0  c
5  4  3  3.0  d
6  4  3  NaN  e
8  6  1  1.0  f
9  4  1  1.0  g
9  3  1  1.0  h
9  7  2  2.0  t

>>> pdf.groupby((10, "a"))[[(20, "c")]].sum()
          20
           c
(10, a)
1        4.0
2        2.0
3        1.0
4        4.0
6        8.0
7        2.0
```

Reply:
Oh, Sorry. nvm above. It's different from `pdf.groupby((10, "a"))[(20, "c")].sum()` anyway.